src/server.js

```diff
@@ -655,6 +656,28 @@ function initializeClient(socket, client, token, lastMessage, openChannel) {
 				socket.emit("search:results", results);
 			});
 		});
+
+		socket.on("mute:change", ({target, setMutedTo}) => {
+			const {chan, network} = client.find(target);
+
+			// If the user mutes the lobby, we mute the entire network.
+			if (chan.type === Chan.Type.LOBBY) {
+				for (const channel of network.channels) {
+					channel.setMuteStatus(setMutedTo);
+				}
+			} else {
+				chan.setMuteStatus(setMutedTo);
+			}
+
+			for (const attachedClient of Object.keys(client.attachedClients)) {
```

Comment:
I think we'll send a `mute:changed` event back down to the client who just sent one to us. Might be best to exclude them?

Reply:
the client (even the one that emits the `mute:change` event) waits for the `mute:changed` acknowledgement from the server, so we want all clients to receive the event. This let me not repeat the 'mute channel' code on the client and just mute channels when the event is received.